src/libsyntax/ext/base.rs

```diff
@@ -725,7 +726,7 @@ impl SyntaxExtension {
             SyntaxExtension::IdentTT { .. } |
             SyntaxExtension::MultiDecorator(..) |
             SyntaxExtension::MultiModifier(..) |
-            SyntaxExtension::BuiltinDerive(..) => hygiene::default_edition(),
+            SyntaxExtension::BuiltinDerive(..) => Edition::from_session(),
```

Comment:
It's hard to search for this, but I suspect that `parse_sess.edition` is available in all places where `SyntaxExtension::edition` is called, so it probably could get an extra parameter `fn edition(&self, default_edition: Edition)`.